### PR TITLE
Tensorflow 1.x backend: layer-by-layer dropout rate setting for DeepONet

### PR DESCRIPTION
**Pull request by vl-dud**
Now you can set `dropout_rate` more flexibly:
```
net = dde.nn.DeepONet(
    [10, 128, 128, 128],
    [dim_x, 128, 128, 128],
    "tanh",
    "Glorot normal",
    dropout_rate={
        "branch": [0.5, 0.3],
        "trunk": [0.5, 0.3, 0.2]
    },
)
```

### CODE REVIEW

**Comment by lululxvi** [comment]
There are a lot of code like this
```
if isinstance(self.dropout_rate_branch, list):
```
To simplify the code, you can convert the single dropout to a list as well. Then in the code, there is no need to use `if` to check.

**Comment by vl-dud** [comment]
Fixed

File: deepxde/nn/tensorflow_compat_v1/deeponet.py [hunk 1]
```diff
@@ -155,7 +155,13 @@ class DeepONet(NN):
             both trunk and branch nets. If `activation` is a ``dict``, then the trunk
             net uses the activation `activation["trunk"]`, and the branch net uses
             `activation["branch"]`.
-        dropout_rate (float): The dropout rate, between 0 and 1.
+        dropout_rate: If `dropout_rate` is a ``float`` between 0 and 1, then the same rate is used in
```
Comment (lululxvi) [inline deepxde/nn/tensorflow_compat_v1/deeponet.py:158]: max 88 characters per line

Comment (vl-dud) [reply deepxde/nn/tensorflow_compat_v1/deeponet.py]: Fixed

File: deepxde/nn/tensorflow_compat_v1/deeponet.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def build(self):
             )
         if branch.shape[-1] % self.net.num_outputs != 0:
             raise AssertionError(
-                f"Output size of the branch net is not evenly divisible by {self.net.num_outputs}."
+                f"Output size of the branch net is not "
```
Comment (lululxvi) [inline deepxde/nn/tensorflow_compat_v1/deeponet.py:83]: To simplify things, don't change the code that is irrelevant to this PR.

Comment (vl-dud) [reply deepxde/nn/tensorflow_compat_v1/deeponet.py]: Fixed

**Pull request merged**